Compilation: error with ARM HPC compiler suite

<!--Thank you for taking the time to submit an issue!-->

## Background information

Compilation error using ARM 22 (+ minor versions)

### What version of the PMIx Reference Library are you using? <!--(e.g., v1.0, v2.1, git master @ hash, etc.)-->

github master @ faed8ee7c5402acfc7940ed83a082593d2a00901

### Describe how PMIx was installed <!--(e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)-->

clone

### Please describe the system on which you are running

* Operating system/version: CentOS Linux release 8.1.1911 (Core)
* Computer hardware: aarch64
* Network type: IBV

-----------------------------

## Details of the problem

```Making all in mca/base
make[2]: Entering directory '/autofs/nccs-svm1_envoy_od/arcurtis/src/pmix/build/src/mca/base'
depbase=`echo pmix_mca_base_var_enum.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/sh ../../../libtool  --tag=CC   --mode=compile armclang -DHAVE_CONFIG_H -I. -I../../../../pmix-git/src/mca/base -I../../../src/include -I../../../include   -iquote/ccsopen/home/arcurtis/src/pmix/build -iquote/ccsopen/home/arcurtis/src/pmix/pmix-git -iquote/ccsopen/home/arcurtis/src/pmix/pmix-git/src -iquote/ccsopen/home/arcurtis/src/pmix/build/include -I/ccsopen/home/arcurtis/src/pmix/pmix-git/include   -I/ccsopen/home/arcurtis/opt/hwloc/2.4.1/include -I/usr/include/libxml2  -O3 -DNDEBUG  -finline-functions -pthread -fvisibility=hidden -Wundef -Wno-long-long -Wsign-compare -Wmissing-prototypes -Wstrict-prototypes -Wcomment -Wshadow -Werror-implicit-function-declaration -fno-strict-aliasing -pedantic -Wall -Wextra -Werror -MT pmix_mca_base_var_enum.lo -MD -MP -MF $depbase.Tpo -c -o pmix_mca_base_var_enum.lo ../../../../pmix-git/src/mca/base/pmix_mca_base_var_enum.c &&\
mv -f $depbase.Tpo $depbase.Plo
libtool: compile:  armclang -DHAVE_CONFIG_H -I. -I../../../../pmix-git/src/mca/base -I../../../src/include -I../../../include -iquote/ccsopen/home/arcurtis/src/pmix/build -iquote/ccsopen/home/arcurtis/src/pmix/pmix-git -iquote/ccsopen/home/arcurtis/src/pmix/pmix-git/src -iquote/ccsopen/home/arcurtis/src/pmix/build/include -I/ccsopen/home/arcurtis/src/pmix/pmix-git/include -I/ccsopen/home/arcurtis/opt/hwloc/2.4.1/include -I/usr/include/libxml2 -O3 -DNDEBUG -finline-functions -pthread -fvisibility=hidden -Wundef -Wno-long-long -Wsign-compare -Wmissing-prototypes -Wstrict-prototypes -Wcomment -Wshadow -Werror-implicit-function-declaration -fno-strict-aliasing -pedantic -Wall -Wextra -Werror -MT pmix_mca_base_var_enum.lo -MD -MP -MF .deps/pmix_mca_base_var_enum.Tpo -c ../../../../pmix-git/src/mca/base/pmix_mca_base_var_enum.c  -fPIC -DPIC -o .libs/pmix_mca_base_var_enum.o
../../../../pmix-git/src/mca/base/pmix_mca_base_var_enum.c:304:9: error: variable 'all_flags' set but not used [-Werror,-Wunused-but-set-variable]
    int all_flags = 0;
        ^
1 error generated.
```

```
$ armclang --version
Arm C/C++/Fortran Compiler version 22.0.2 (build number 1776) (based on LLVM 13.0.0)
Target: aarch64-unknown-linux-gnu
Thread model: posix
InstalledDir: /autofs/nccs-svm1_wombat_sw/ARM_Compiler/22.0.2/arm-linux-compiler-22.0.2_Generic-AArch64_RHEL-8_aarch64-linux/bin
```

--- a/src/mca/base/pmix_mca_base_var_enum.c
+++ b/src/mca/base/pmix_mca_base_var_enum.c
@@ -301,7 +301,9 @@ int pmix_mca_base_var_enum_create_flag(const char *name,
         return PMIX_ERR_OUT_OF_RESOURCE;
     }
 
+#if PMIX_ENABLE_DEBUG
     int all_flags = 0;
+#endif
     for (i = 0; i < new_enum->super.enum_value_count; ++i) {
         new_enum->enum_flags[i].flag = flags[i].flag;
         new_enum->enum_flags[i].string = strdup(flags[i].string);
@@ -312,7 +314,9 @@ int pmix_mca_base_var_enum_create_flag(const char *name,
         assert(!(flags[i].flag & flags[i].conflicting_flag));
         assert(!(all_flags & flags[i].flag));
         assert(flags[i].flag);
+#if PMIX_ENABLE_DEBUG
         all_flags |= flags[i].flag;
+#endif
     }
 
     *enumerator = new_enum;

--- a/src/mca/bfrops/base/bfrop_base_unpack.c
+++ b/src/mca/bfrops/base/bfrop_base_unpack.c
@@ -54,8 +54,9 @@ static pmix_status_t pmix_bfrops_base_unpack_buffer(pmix_pointer_array_t *regtyp
         }
         /* if the data types don't match, then return an error */
         if (type != local_type) {
-            pmix_output(0, "PMIX bfrop:unpack: got type %d when expecting type %d", local_type,
-                        type);
+            pmix_output(0, "PMIX bfrop:unpack: got type %s when expecting type %s",
+                        PMIx_Data_type_string(local_type),
+                        PMIx_Data_type_string(type));
             return PMIX_ERR_PACK_MISMATCH;
         }
     }

